resources/sdk/clisdkclient/extensions/utils/filter.go

```diff
@@ -0,0 +1,270 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mypurecloud/platform-client-sdk-cli/build/gc/logger"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	TypeInteger string = "int"
+	TypeString         = "string"
+	TypeBoolean        = "bool"
+	TypeArray          = "array"
+)
+
+const (
+	equalsOperator            string = "=="
+	notEqualsOperator                = "!="
+	lessThanEqualsOperator           = "<="
+	greaterThanEqualsOperator        = ">="
+	lessThanOperator                 = "<"
+	greaterThanOperator              = ">"
+	containsOperator                 = "contains"
+	matchOperator                    = "match"
+)
+
+var (
+	supportedOperators = []string{
+		equalsOperator,
+		notEqualsOperator,
+		lessThanEqualsOperator,
+		greaterThanEqualsOperator,
+		lessThanOperator,
+		greaterThanOperator,
+		containsOperator,
+		matchOperator,
+	}
+)
+
+func FilterByCondition(data string, condition string) (string, error) {
+	objects, err := getJsonObjectsFromString(data)
+	if err != nil {
+		return "", err
+	}
+
+	// Find operator in condition
+	operator := findOperatorInString(condition)
+	if operator == "" {
+		return "", unrecognizedConditionError(condition)
+	}
+
+	path, value := getFieldKeyAndValueFromConditionString(condition, operator)
+	keys := getKeysFromJsonFieldPath(path)
+
+	allMatchedObjects, err := findObjectsMatchingCondition(objects, keys, value, operator)
+	if err != nil {
+		return "", err
+	}
+
+	// Convert go data to json byte array
+	jsonBytes, err := json.MarshalIndent(allMatchedObjects, "", "  ")
+	if err != nil {
+		return "", err
+	}
+
+	return string(jsonBytes), nil
+}
+
+func findObjectsMatchingCondition(returnedObjects []interface{}, keys []string, value string, operator string) ([]interface{}, error) {
+	var (
+		allMatchedObjects []interface{}
+		currentObject     interface{}
+		match             bool
+		err               error
+	)
+
+	for _, object := range returnedObjects {
+		currentObject = object
+		for i, k := range keys {
+			if currentObjectMap, ok := currentObject.(map[string]interface{}); ok {
```

Comment:
Hi Charlie,

I would refactor the code below to get rid of the if..then..else.  Properly structured go will do a | line down from each if..then statement.  So you check something and grab the data and return rather then doing a nest of if..then statements.

Reply:
@carnellj-genesys  done ✅